src/main/java/Duke.java

```diff
@@ -1,10 +1,93 @@
+import java.util.Scanner;
+
 public class Duke {
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
+        int count = 0;
+        String input;
+        Task list[] = new Task[100];
+
+        printWelcomeMessage();
+
+        // Echo loops until "bye" is inputted
+        while(true) {
+            Scanner echo = new Scanner(System.in); // Scan and echo user input
+            input = echo.nextLine();
+
+            // Exit program if "bye" is inputted
+            if (input.equals("bye")) {
+                System.out.println("Goodbye. See you soon!");
+                break;
+            }
+
+            else if (input.equals("list")){
+                printList(list);
+            }
+
+            else if (input.contains("done")){
+                markAsDone(input, list);
+            }
+
+            // Add input as new task
```

Comment:
I like how you add an additional comment here to annotate the function of this code block.